server/src/main/java/io/druid/server/lookup/cache/LookupCoordinatorManager.java

```diff
@@ -506,137 +257,398 @@ public boolean deleteLookup(final String tier, final String lookup, AuditInfo au
-      executorService.shutdownNow();
-      final ListenableScheduledFuture backgroundManagerFuture = this.backgroundManagerFuture;
-      this.backgroundManagerFuture = null;
-      if (backgroundManagerFuture != null && !backgroundManagerFuture.cancel(true)) {
-        LOG.warn("Background lookup manager thread could not be cancelled");
+
+      Set<String> toDrop = new HashSet<>();
+      toDrop.addAll(lookupsState.getCurrent().keySet());
+      toDrop.addAll(lookupsState.getToLoad().keySet());
+      toDrop = Sets.difference(toDrop, lookupsState.getToDrop());
+      toDrop = Sets.difference(toDrop, nodeTierLookups.keySet());
+
+      if (!toLoad.isEmpty() || !toDrop.isEmpty()) {
+        stateToUpdate.put(node, updateNode(node, new LookupsStateWithMap(null, toLoad, toDrop)));
+
+        LOG.debug(
+            "Sent lookup toAdd[%d] and toDrop[%d] updates to node [%s].",
+            toLoad.size(),
+            toDrop.size(),
+            node
+        );
+      }
+
+      LOG.debug("Finished lookup sync for node [%s].", node);
+    }
+    catch (Exception ex) {
+      LOG.makeAlert(ex, "Failed to manage lookups on node [%s].", node).emit();
+    }
+  }
+
+  @VisibleForTesting
+  LookupsStateWithMap updateNode(
+      HostAndPort node,
+      LookupsStateWithMap lookupsUpdate
+  )
+      throws IOException, InterruptedException, ExecutionException
+  {
+    final AtomicInteger returnCode = new AtomicInteger(0);
+    final AtomicReference<String> reasonString = new AtomicReference<>(null);
+
+    final URL url = getLookupsUpdateURL(node);
+
+    LOG.debug("Sending lookups load/drop request to [%s]. Request [%s]", url, lookupsUpdate);
+
+    try (final InputStream result = httpClient.go(
+        new Request(HttpMethod.POST, url)
+            .addHeader(HttpHeaders.Names.ACCEPT, SmileMediaTypes.APPLICATION_JACKSON_SMILE)
+            .addHeader(HttpHeaders.Names.CONTENT_TYPE, SmileMediaTypes.APPLICATION_JACKSON_SMILE)
+            .setContent(smileMapper.writeValueAsBytes(lookupsUpdate)),
+        makeResponseHandler(returnCode, reasonString),
+        lookupCoordinatorManagerConfig.getHostTimeout()
+    ).get()) {
+      if (httpStatusIsSuccess(returnCode.get())) {
+        try {
+          final LookupsStateWithMap response = smileMapper.readValue(result, LookupsStateWithMap.class);
+          LOG.debug(
+              "Update on [%s], Status: %s reason: [%s], Response [%s].", url, returnCode.get(), reasonString.get(),
+              response
+          );
+          return response;
+        } catch (IOException ex) {
+          throw new IOException(
+              String.format("Failed to parse update response from [%s]. response [%s]", url, result),
+              ex
+          );
+        }
+      } else {
+        final ByteArrayOutputStream baos = new ByteArrayOutputStream();
+        try {
+          StreamUtils.copyAndClose(result, baos);
+        }
+        catch (IOException e2) {
+          LOG.warn(e2, "Error reading response");
```

Comment:
this exception only would say that failed to read InputStream ... later exception actually has meaningful information.... end result is same, they both get printed in the log.

btw, this is a copy paste of your old code so must have thought this was good enough :)